### PR TITLE
Fix error 500 for Visits URL within time Point action

### LINKED ISSUE #10316: Visits specific URL point action issues
## Bug Description
When new contact is created through the tracking script on external website and you have published "Visits specific URL" point action with "Returns within/after" option then the points will be added on the first hit.

| Q   | A
| --- | ---
| Mautic version | 4.0.0-rc and previous
| PHP version | 7.4
| Browser | Any / tested on FF 90.0.2

### Steps to reproduce
1. Create and publish "Visits specific URL" point action with "Returns within" parameter
obraz
2. Go to website specified in the action as a new visitor (browser private window or new device). The website must have your mtc  tracking script. 
3. Your new anonymous contact has 1 page visit in history and extra points
obraz

 
### Log errors
Any configuration with "Visits specific URL" generates huge number or logs when you are using tracking script on websites:
```
[2021-08-06 08:56:34] mautic.NOTICE: PHP Notice - Trying to access array offset on value of type bool - in file /srv/mautic/app/bundles/PageBundle/Entity/HitRepository.php - at line 276 {"options":{"leadId":"10","0":"http://unserializer.local/","second_to_last":"15"},"sq":"[object] (Doctrine\\DBAL\\Query\\QueryBuilder: SELECT h.date_hit latest_hit FROM page_hits h WHERE (h.lead_id = 10) AND (h.id <> 15))","result":false} []
```

The error is caused by `\Mautic\PageBundle\Entity\HitRepository::getLatestHit` method because it returns current date object when the latest hit is not found. This object is used in the comparison in the "returns_within/returns_after" condition.

### PR DESCRIPTION
**Pull request by patrykgruszka**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ Y ]
| New feature/enhancement? (use the a.x branch)      | [ N ]
| Deprecations?                          | [ N ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ Y ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #10316 

#### Description:
When new contact is created through the tracking script on external website and you have published "Visits specific URL" point action with "Returns within/after" the Mautic respone will return status 500.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create and publish "Visits specific URL" point action with "Returns within" parameter
image
3. OPTION 1:Go to website specified in the action as a new visitor (browser private window or new device). The website must have your mtc tracking script.
OPTION 2: Simulate the request from website with curl. You may need to replace URLs in the curl request
```
curl 'https://mautic.ddev.site/mtc/event' -X POST \
  -H 'User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/116.0' \
  -H 'Accept: */*' \
  -H 'Accept-Language: en-US,en;q=0.5' \
  -H 'Accept-Encoding: gzip, deflate, br' \
  -H 'Referer: https://yourwebsite.example.com' \
  -H 'Content-Type: application/x-www-form-urlencoded' \
  -H 'X-Requested-With: XMLHttpRequest' \
  -H 'Origin: https://yourwebsite.example.com' \
  -H 'DNT: 1' \
  -H 'Connection: keep-alive' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: cross-site' \
  --data-raw 'page_title=TEST_TITLE&page_language=en-US&preferred_locale=en_US&page_referrer=&page_url=https%3A%2F%2Fyourwebsite.example.com&counter=0&timezone_offset=-120&resolution=1920x1080&platform=UNIX&do_not_track=true&timezone=Europe%2FWarsaw'
```
4. The request should return a success status (200) and an anonymous contact should be created with a page view:
image
5. Points for action should not be assigned at this point, as they will be assigned when the contact returns to the page.

### CODE REVIEW

File: app/bundles/PageBundle/Entity/HitRepository.php [hunk 1]
```diff
@@ -232,7 +232,7 @@ public function countVisitors($seconds = 60, $notLeft = false)
      *
      * @param array $options
      *
-     * @return \DateTime
+     * @return \DateTime|null
      */
     public function getLatestHit($options)
```
Comment (tomekkowalczyk) [inline app/bundles/PageBundle/Entity/HitRepository.php]: You could use return type from methods `public function getLatestHit($options): ?\DateTime` and delete the comment

**Review by tomekkowalczyk — APPROVED** [review]

File: app/bundles/PageBundle/Entity/HitRepository.php (hunk 1 above)
Comment (patrykgruszka) [reply app/bundles/PageBundle/Entity/HitRepository.php]: Changing the signature of the public method would be a breaking change, which I would rather avoid in this case.

**Review by magdalenaleonow — APPROVED** [review]
Tested, everything works:
image

**Review by kuzmany — APPROVED** [review]
Code looks good 👍

**Pull request merged**